### DISCUSSION TITLE
There was no runtime pack for Microsoft.WindowsDesktop.App available for the specified RuntimeIdentifier 'win-arm64'.

### DISCUSSION THREAD

**Discussion by limeniye** [unoplatform/uno#13028] [Q&A] [answered]
<h1>Problem</h1>

I`am trying to build my shared UnoApplication project:
![image](https://github.com/unoplatform/uno/assets/69314237/f0f45f59-fdc1-4a58-adb8-0c0f5dd611e5)

But at the output I have the following message:
```
1>C:\Program Files\dotnet\sdk\8.0.100-preview.6.23330.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(479,5): error NETSDK1082: There was no runtime pack for Microsoft.NETCore.App available for the specified RuntimeIdentifier 'win10-arm'.
1>C:\Program Files\dotnet\sdk\8.0.100-preview.6.23330.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(479,5): error NETSDK1082: There was no runtime pack for Microsoft.NETCore.App available for the specified RuntimeIdentifier 'win10-arm-aot'.
```

<h1>What I already tried to do</h1>

<h3>Uno-check</h3>

I started uno-check (1.13.0.0)
```
uno-check --preview-major
```

There I have the following message:
![image](https://github.com/unoplatform/uno/assets/69314237/93c1fd6d-60a6-405a-b083-a8d153b3b2f1)

But uno-check not fixing this problem.

<h4>Option info</h4>

My ```C:\Windows\System32\global.json```
```JSON
{
  "sdk": {
    "version": "8.0.100-preview.6.23330.14",
    "allowPrerelease": true
  }
}
```
My ```dotnet --list-sdks```
```
7.0.400-preview.23330.10 [C:\Program Files\dotnet\sdk]
8.0.100-preview.2.23157.25 [C:\Program Files\dotnet\sdk]
8.0.100-preview.6.23330.14 [C:\Program Files\dotnet\sdk]
```
<h3>RuntimeIdentifiers</h3>
I added the following lines to my .csproj file

```XML
<PropertyGroup Condition="'$(TargetFramework)' == 'net8.0-windows10.0.19041'">
  <RuntimeIdentifiers>win10-arm;win10-arm-aot;</RuntimeIdentifiers>    
</PropertyGroup>
```
It didn`t helps me.

<h1>Info</h1>

Visual Studio Community 2022 17.7.0 Preview 5.0
Windows 11
Uno.WinUI 4.9.17

**Comment by jeromelaban** [comment]
Thanks for the report. This looks like a WinAppSDK issue. Does a standard WinAppSDK library build properly?

**Reply by limeniye** [reply]
To build WinAppSDK I cloned this repo: https://github.com/microsoft/WindowsAppSDK

It asked me to login:
![image](https://github.com/unoplatform/uno/assets/69314237/66a4d741-eb88-4976-862c-c17d9fb79701)

I did it and it was successfull. But this window are popuping again and again.
I can`t build this this library.

```
5>C:\Repositories\WindowsAppSDK\Directory.Build.props(33,5): error : C:\Repositories\WindowsAppSDK\.user\winappsdk.certificate.test.pfx not found. Run 'C:\Repositories\WindowsAppSDK\dev\Deployment\..\..\\DevCheck.cmd' to generate the test certificate.
```
I runned this cmd scrip:

```
PS C:\Repositories\temp> C:\Repositories\WindowsAppSDK\DevCheck.cmd
Checking developer environment...
Windows App SDK location...C:\Repositories\WindowsAppSDK
VisualStudio 2022...
Creating C:\Repositories\WindowsAppSDK\.user...
Test certificate thumbprint C:\Repositories\WindowsAppSDK\.user\winappsdk.certificate.test.thumbprint...Not Found
Test certificate .pfx...Access Denied. Run from an admin prompt
Fatal error. Aborting...
PS C:\Repositories\temp>
```

**Reply by jeromelaban** [reply]
You did not have to go this far. I'm simply asking to create a class library using the WinAppSDK templates, and build it using `net8.0-windows` to see if the same issue happens.

**Reply by limeniye** [reply]
I created a simple WinUI class library. It builds well.

I copied following line from WinUI class library ```<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>``` to Uno class library:

```XML
	<PropertyGroup Condition="'$(TargetFramework)' == 'net8.0-windows10.0.19041'">
		<!--<RuntimeIdentifiers>win10-arm;win10-arm-aot;</RuntimeIdentifiers>-->
		<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
	</PropertyGroup>
```

The compiler said me the following message:
![image](https://github.com/unoplatform/uno/assets/69314237/668fb9ed-71dc-43ba-ba6a-9dfe2a5e005f)


I fixed it by the following way:
```XML
<UnoUIUseRoslynSourceGenerators>true</UnoUIUseRoslynSourceGenerators>
```

![image](https://github.com/unoplatform/uno/assets/69314237/becf0135-ef38-4caf-9b15-85d7f602e59c)

Currently I don't know which is real problem is.

**Reply by limeniye** [reply]
Without this line:
```XML
<RuntimeIdentifiers>win10-x86;win10-x64;win10-arm64</RuntimeIdentifiers>
```
it doesnt work and say me the following message:
```
1>C:\Program Files\dotnet\sdk\8.0.100-preview.6.23330.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(479,5): error NETSDK1082: There was no runtime pack for Microsoft.NETCore.App available for the specified RuntimeIdentifier 'win10-arm'.
1>C:\Program Files\dotnet\sdk\8.0.100-preview.6.23330.14\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(479,5): error NETSDK1082: There was no runtime pack for Microsoft.NETCore.App available for the specified RuntimeIdentifier 'win10-arm-aot'.
```

**Reply by jeromelaban** [reply, marked as answer]
Thanks for the updates, let's track in https://github.com/unoplatform/uno.templates/issues/183
